code/modules/power/engines/tesla/energy_ball.dm

```diff
@@ -117,28 +121,104 @@
 		var/turf/T = get_step(src, pick(GLOB.alldirs))
 		if(can_move(T))
 			forceMove(T)
-			has_close_field = FALSE
 			for(var/mob/living/carbon/C in loc)
 				dust_mobs(C)
 		return
-	if(!target_turf)
+
+	if(!target_turf || loc == target_turf)
 		find_the_basket()
 		return
-	for(var/i in 0 to 8)
-		movement_dir = get_dir(get_turf(src), target_turf)
-		forceMove(get_step(src, movement_dir))
-		if(get_turf(src) == target_turf)
-			target_turf = null
-		for(var/mob/living/carbon/C in loc)
-			dust_mobs(C)
-		has_close_field = FALSE
 
+	var/turf/move_target = next_move_target()
+	movement_beam(move_target, 1.5 SECONDS)
+	sleep(0.5 SECONDS)
+	// MORE POWER
+	movement_beam(move_target, 1 SECONDS)
+	sleep(0.5 SECONDS)
+	walk_towards(src, move_target, 0, 10)
+
+/datum/move_with_corner
+	var/turf/start
+	var/turf/end
+	var/turf/corner
+	var/pre_corner_x_unit
+	var/pre_corner_y_unit
+	var/pre_corner_dist
+	var/post_corner_x_unit
+	var/post_corner_y_unit
+	var/post_corner_dist
+
+/datum/move_with_corner/New(atom/start_in, atom/end_in)
+	start = start_in
+	end = end_in
+	pre_corner_x_unit = (end.x > start.x) ? 1 : -1
+	pre_corner_y_unit = (end.y > start.y) ? 1 : -1
+	if(start.x == end.x)
+		// Moving along y
+		corner = end
+		pre_corner_x_unit = 0
+		pre_corner_dist = abs(start.y - end.y)
```

Comment:
Would it be feasible to calculate the absolute between the start and end positions once and just store and reuse that variable?

Reply:
Discussed via Discord. Resolving.